### ISSUE TITLE
Example with "writing 10 messages, then moving to the 5th" fails when moving accross rolled files

### ISSUE THREAD

**Issue by abysas** [OpenHFT/Chronicle-Queue#598]
The example presented at https://github.com/OpenHFT/Chronicle-Queue#how-to-write-10-messages-then-move-to-the-5th-message-to-read-it fails if I set RollCycles.TEST_SECONDLY and introduce 1.5 s delay between appends in the loop:
```
    public void read5thMessageTest() throws MalformedURLException, InterruptedException {
        try (final ChronicleQueue queue = ChronicleQueue.singleBuilder(getTmpDir()).rollCycle(RollCycles.TEST_SECONDLY).build()) {

            final ExcerptAppender appender = queue.acquireAppender();

            int i = 0;
            String msg = "world ";
            for (int j = 0; j < 10; j++) {

                try (DocumentContext dc = appender.writingDocument()) {
                    dc.wire().write("hello").text(msg + (i++));
                    long indexWritten = dc.index();
                }
                Thread.sleep(1500);
            }

            // get the current cycle
            int cycle;
            final ExcerptTailer tailer = queue.createTailer();
            try (DocumentContext documentContext = tailer.readingDocument()) {
                long index = documentContext.index();
                cycle = queue.rollCycle().toCycle(index);
            }

            long index = queue.rollCycle().toIndex(cycle, 5);
            tailer.moveToIndex(index);
            try (DocumentContext dc = tailer.readingDocument()) {
                String s5 = dc.wire().read("hello").text();
                System.out.println(s5);
                assertEquals(msg + 5, s5);
            }
        }
    }
```

The assertion fails:
```
org.opentest4j.AssertionFailedError: 
Expected :world 5
Actual   :world 1
```

**Comment by abysas** [comment]
I reproduced the same issue when roll minutely and increased the delay in the loop to 7 s. Seems that moving to index across the rolled files are not working correctly. If it is by design, could then the example be corrected to show the safe way on moving to _n_th item in the queue.

Chronicle Queue v5.17.10
OS: Windows 10 x64

**Comment by RobAustin** [comment]
The test should be change to use the time provider rather than pausing 1.5 seconds

**Comment by abysas** [comment]
Could you please provide more infornation on using the time provider? What is it?

Technicaly append may occur at any time. How then this differ from Thread.sleep() while being alligned with "using time provider"?

**Comment by abysas** [comment]
@RobAustin, could you please elaborate a bit on your suggestion about the time provider and how it can be related to appender adding entries at random time? 
In the present example the delay is caused by `Thread.sleep()`, but equivalent blocking could be caused by any other slow resource. To us it is a blocker issue as we cannot reliably move to particular message sequence.

**Comment by RobAustin** [comment]
For use of the time provider, see other tests in chronicle-queue which use the time provider. The time provider let’s the test control the time that actions are taken. Making these timing related test totally reproducible. This is better than relying on a sleep for a certain number of milliseconds which may or may not return by that time.

**Comment by RobAustin** [comment]
If you have a blocker issue and you wish us to give it priority. Please contact sales@chronicle.software and we can discuss the support contracts we offer.

**Comment by peter-lawrey** [comment]
In this case, `moveToIndex` is returning false and you only have one message in each cycle. i.e. there is no 5th message in any cycle.  I have fixed it so when moveToIndex returns false, it won't just read the entry it would have read before this call.
I have added how you can read forward 5 messages from a given point.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.20.42](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.20.42), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)
